app/Repositories/Organization/OrganizationRepository.php

```diff
@@ -0,0 +1,413 @@
+    {
+        return $organization->users()->withCount([
+            'projects' => function ($query) use ($organization) {
+                $query->where('organization_id', $organization->id);
+            },
+            'teams' => function ($query) use ($organization) {
+                $query->where('organization_id', $organization->id);
+            },
+            'groups' => function ($query) use ($organization) {
+                $query->where('organization_id', $organization->id);
+            }
+        ])
+        ->when($data['query'] ?? null, function ($query) use ($data) {
+            $query->where('email', 'like', '%' . $data['query'] . '%');
+            return $query;
+        })
+        ->paginate();
+    }
+
+    /**
+     * Get admin
+     *
+     * @param Organization $organization
+     * @return Model
+     */
+    public function getAdmin($organization)
+    {
+        try {
+            if ($organization) {
+                return $organization->users()->wherePivot('organization_role_type_id', 1)->first();
+            }
+        } catch (\Exception $e) {
+            Log::error($e->getMessage());
+        }
+    }
+
+    /**
+     * Invite member to the organization
+     *
+     * @param User $authenticatedUser
+     * @param Organization $organization
+     * @param $data
+     * @return bool
+     */
+    public function inviteMember($authenticatedUser, $organization, $data)
+    {
+        $invited = true;
+
+        $user = $this->userRepository->findByField('email', $data['email']);
+        $note = array_key_exists('note', $data) ? $data['note'] : '';
+
+        if ($user) {
+            $organization->users()->syncWithoutDetaching([$user->id => ['organization_role_type_id' => $data['role_id']]]);
+            $user->notify(new OrganizationInvite($authenticatedUser, $organization, 'login', $note));
+        } elseif ($data['email']) {
+            $token = Hash::make((string)Str::uuid() . date('D M d, Y G:i'));
+            $temp_user = new User(['email' => $data['email']]);
+            $temp_user->notify(new OrganizationInvite($authenticatedUser, $organization, 'register', $note));
+
+            $invited_user = array(
+                'invited_email' => $data['email'],
+                'organization_id' => $organization->id,
+                'organization_role_type_id' => $data['role_id'],
+                'token' => $token,
+            );
+            $this->invitedOrganizationUserRepository->create($invited_user);
+        } else {
+            $invited = false;
+        }
+
+        return $invited;
+    }
+
+    /**
+     * To fetch organization user permissions
+     *
+     * @param User $authenticatedUser
+     * @param Organization $organization
+     * @return Model
+     */
+    public function fetchOrganizationUserPermissions($authenticatedUser, $organization)
+    {
+        $orgUsrPermissions = $organization->userRoles()
+                ->wherePivot('user_id', $authenticatedUser->id)
```

Comment:
Add this function in the interface as well.
All custom functions should be added in the interface as well.

Reply:
Added "fetchOrganizationUserPermissions()" to OrganizationRepositoryInterface.